In [11]:
import pandas as pd
import matplotlib.pyplot as mat
import seaborn as sea

#### Importing the libraries pandas ,matplotlib.pyplot an seaborn from python, in order to perform column operations on the dataset.


In [8]:
market=pd.read_csv("C:/Users/anand/OneDrive/Desktop/super_market.csv")

In [9]:
market

Invoice ID Branch       City Customer type  Gender  \
0    750-67-8428      A     Yangon        Member  Female   
1    226-31-3081      C  Naypyitaw        Normal  Female   
2    631-41-3108      A     Yangon        Normal    Male   
3    123-19-1176      A     Yangon        Member    Male   
4    373-73-7910      A     Yangon        Normal    Male   
..           ...    ...        ...           ...     ...   
995  233-67-5758      C  Naypyitaw        Normal    Male   
996  303-96-2227      B   Mandalay        Normal  Female   
997  727-02-1313      A     Yangon        Member    Male   
998  347-56-2442      A     Yangon        Normal    Male   
999  849-09-3807      A     Yangon        Member  Female   

               Product line  Unit price  Quantity   Tax 5%      Total  \
0         Health and beauty       74.69         7  26.1415   548.9715   
1    Electronic accessories       15.28         5   3.8200    80.2200   
2        Home and lifestyle       46.33         7  16.2155   340.5255   
3         Health and beauty       58.22         8  23.2880   489.0480   
4         Sports and travel       86.31         7  30.2085   634.3785   
..                      ...         ...       ...      ...        ...   
995       Health and beauty       40.35         1   2.0175    42.3675   
996      Home and lifestyle       97.38        10  48.6900  1022.4900   
997      Food and beverages       31.84         1   1.5920    33.4320   
998      Home and lifestyle       65.82         1   3.2910    69.1110   
999     Fashion accessories       88.34         7  30.9190   649.2990   

          Date   Time      Payment    cogs  gross margin percentage  \
0     1/5/2019  13:08      Ewallet  522.83                 4.761905   
1     3/8/2019  10:29         Cash   76.40                 4.761905   
2     3/3/2019  13:23  Credit card  324.31                 4.761905   
3    1/27/2019  20:33      Ewallet  465.76                 4.761905   
4     2/8/2019  10:37      Ewallet  604.17                 4.761905   
..         ...    ...          ...     ...                      ...   
995  1/29/2019  13:46      Ewallet   40.35                 4.761905   
996   3/2/2019  17:16      Ewallet  973.80                 4.761905   
997   2/9/2019  13:22         Cash   31.84                 4.761905   
998  2/22/2019  15:33         Cash   65.82                 4.761905   
999  2/18/2019  13:28         Cash  618.38                 4.761905   

     gross income  Rating  
0         26.1415     9.1  
1          3.8200     9.6  
2         16.2155     7.4  
3         23.2880     8.4  
4         30.2085     5.3  
..            ...     ...  
995        2.0175     6.2  
996       48.6900     4.4  
997        1.5920     7.7  
998        3.2910     4.1  
999       30.9190     6.6  

[1000 rows x 17 columns]

In [13]:
market.columns

Index(['Invoice ID', 'Branch', 'City', 'Customer type', 'Gender',
       'Product line', 'Unit price', 'Quantity', 'Tax 5%', 'Total', 'Date',
       'Time', 'Payment', 'cogs', 'gross margin percentage', 'gross income',
       'Rating'],
      dtype='object')

#### Displaying the variable names in the dataset.

In [16]:
market['total']=market['Unit price']+market['Quantity']

#### Added a new variable in the dataset by combining the variables Unit price and Quantity.

In [17]:
market

Invoice ID Branch       City Customer type  Gender  \
0    750-67-8428      A     Yangon        Member  Female   
1    226-31-3081      C  Naypyitaw        Normal  Female   
2    631-41-3108      A     Yangon        Normal    Male   
3    123-19-1176      A     Yangon        Member    Male   
4    373-73-7910      A     Yangon        Normal    Male   
..           ...    ...        ...           ...     ...   
995  233-67-5758      C  Naypyitaw        Normal    Male   
996  303-96-2227      B   Mandalay        Normal  Female   
997  727-02-1313      A     Yangon        Member    Male   
998  347-56-2442      A     Yangon        Normal    Male   
999  849-09-3807      A     Yangon        Member  Female   

               Product line  Unit price  Quantity   Tax 5%      Total  \
0         Health and beauty       74.69         7  26.1415   548.9715   
1    Electronic accessories       15.28         5   3.8200    80.2200   
2        Home and lifestyle       46.33         7  16.2155   340.5255   
3         Health and beauty       58.22         8  23.2880   489.0480   
4         Sports and travel       86.31         7  30.2085   634.3785   
..                      ...         ...       ...      ...        ...   
995       Health and beauty       40.35         1   2.0175    42.3675   
996      Home and lifestyle       97.38        10  48.6900  1022.4900   
997      Food and beverages       31.84         1   1.5920    33.4320   
998      Home and lifestyle       65.82         1   3.2910    69.1110   
999     Fashion accessories       88.34         7  30.9190   649.2990   

          Date   Time      Payment    cogs  gross margin percentage  \
0     1/5/2019  13:08      Ewallet  522.83                 4.761905   
1     3/8/2019  10:29         Cash   76.40                 4.761905   
2     3/3/2019  13:23  Credit card  324.31                 4.761905   
3    1/27/2019  20:33      Ewallet  465.76                 4.761905   
4     2/8/2019  10:37      Ewallet  604.17                 4.761905   
..         ...    ...          ...     ...                      ...   
995  1/29/2019  13:46      Ewallet   40.35                 4.761905   
996   3/2/2019  17:16      Ewallet  973.80                 4.761905   
997   2/9/2019  13:22         Cash   31.84                 4.761905   
998  2/22/2019  15:33         Cash   65.82                 4.761905   
999  2/18/2019  13:28         Cash  618.38                 4.761905   

     gross income  Rating   total  
0         26.1415     9.1   81.69  
1          3.8200     9.6   20.28  
2         16.2155     7.4   53.33  
3         23.2880     8.4   66.22  
4         30.2085     5.3   93.31  
..            ...     ...     ...  
995        2.0175     6.2   41.35  
996       48.6900     4.4  107.38  
997        1.5920     7.7   32.84  
998        3.2910     4.1   66.82  
999       30.9190     6.6   95.34  

[1000 rows x 18 columns]

In [18]:
market['low']=market ['Unit price']/market['Quantity']

In [20]:
market

Invoice ID Branch       City Customer type  Gender  \
0    750-67-8428      A     Yangon        Member  Female   
1    226-31-3081      C  Naypyitaw        Normal  Female   
2    631-41-3108      A     Yangon        Normal    Male   
3    123-19-1176      A     Yangon        Member    Male   
4    373-73-7910      A     Yangon        Normal    Male   
..           ...    ...        ...           ...     ...   
995  233-67-5758      C  Naypyitaw        Normal    Male   
996  303-96-2227      B   Mandalay        Normal  Female   
997  727-02-1313      A     Yangon        Member    Male   
998  347-56-2442      A     Yangon        Normal    Male   
999  849-09-3807      A     Yangon        Member  Female   

               Product line  Unit price  Quantity   Tax 5%      Total  \
0         Health and beauty       74.69         7  26.1415   548.9715   
1    Electronic accessories       15.28         5   3.8200    80.2200   
2        Home and lifestyle       46.33         7  16.2155   340.5255   
3         Health and beauty       58.22         8  23.2880   489.0480   
4         Sports and travel       86.31         7  30.2085   634.3785   
..                      ...         ...       ...      ...        ...   
995       Health and beauty       40.35         1   2.0175    42.3675   
996      Home and lifestyle       97.38        10  48.6900  1022.4900   
997      Food and beverages       31.84         1   1.5920    33.4320   
998      Home and lifestyle       65.82         1   3.2910    69.1110   
999     Fashion accessories       88.34         7  30.9190   649.2990   

          Date   Time      Payment    cogs  gross margin percentage  \
0     1/5/2019  13:08      Ewallet  522.83                 4.761905   
1     3/8/2019  10:29         Cash   76.40                 4.761905   
2     3/3/2019  13:23  Credit card  324.31                 4.761905   
3    1/27/2019  20:33      Ewallet  465.76                 4.761905   
4     2/8/2019  10:37      Ewallet  604.17                 4.761905   
..         ...    ...          ...     ...                      ...   
995  1/29/2019  13:46      Ewallet   40.35                 4.761905   
996   3/2/2019  17:16      Ewallet  973.80                 4.761905   
997   2/9/2019  13:22         Cash   31.84                 4.761905   
998  2/22/2019  15:33         Cash   65.82                 4.761905   
999  2/18/2019  13:28         Cash  618.38                 4.761905   

     gross income  Rating   total        low  
0         26.1415     9.1   81.69  10.670000  
1          3.8200     9.6   20.28   3.056000  
2         16.2155     7.4   53.33   6.618571  
3         23.2880     8.4   66.22   7.277500  
4         30.2085     5.3   93.31  12.330000  
..            ...     ...     ...        ...  
995        2.0175     6.2   41.35  40.350000  
996       48.6900     4.4  107.38   9.738000  
997        1.5920     7.7   32.84  31.840000  
998        3.2910     4.1   66.82  65.820000  
999       30.9190     6.6   95.34  12.620000  

[1000 rows x 19 columns]

In [21]:
market.rename(columns={'Unit price':'Quantity'})

Invoice ID Branch       City Customer type  Gender  \
0    750-67-8428      A     Yangon        Member  Female   
1    226-31-3081      C  Naypyitaw        Normal  Female   
2    631-41-3108      A     Yangon        Normal    Male   
3    123-19-1176      A     Yangon        Member    Male   
4    373-73-7910      A     Yangon        Normal    Male   
..           ...    ...        ...           ...     ...   
995  233-67-5758      C  Naypyitaw        Normal    Male   
996  303-96-2227      B   Mandalay        Normal  Female   
997  727-02-1313      A     Yangon        Member    Male   
998  347-56-2442      A     Yangon        Normal    Male   
999  849-09-3807      A     Yangon        Member  Female   

               Product line  Quantity  Quantity   Tax 5%      Total  \
0         Health and beauty     74.69         7  26.1415   548.9715   
1    Electronic accessories     15.28         5   3.8200    80.2200   
2        Home and lifestyle     46.33         7  16.2155   340.5255   
3         Health and beauty     58.22         8  23.2880   489.0480   
4         Sports and travel     86.31         7  30.2085   634.3785   
..                      ...       ...       ...      ...        ...   
995       Health and beauty     40.35         1   2.0175    42.3675   
996      Home and lifestyle     97.38        10  48.6900  1022.4900   
997      Food and beverages     31.84         1   1.5920    33.4320   
998      Home and lifestyle     65.82         1   3.2910    69.1110   
999     Fashion accessories     88.34         7  30.9190   649.2990   

          Date   Time      Payment    cogs  gross margin percentage  \
0     1/5/2019  13:08      Ewallet  522.83                 4.761905   
1     3/8/2019  10:29         Cash   76.40                 4.761905   
2     3/3/2019  13:23  Credit card  324.31                 4.761905   
3    1/27/2019  20:33      Ewallet  465.76                 4.761905   
4     2/8/2019  10:37      Ewallet  604.17                 4.761905   
..         ...    ...          ...     ...                      ...   
995  1/29/2019  13:46      Ewallet   40.35                 4.761905   
996   3/2/2019  17:16      Ewallet  973.80                 4.761905   
997   2/9/2019  13:22         Cash   31.84                 4.761905   
998  2/22/2019  15:33         Cash   65.82                 4.761905   
999  2/18/2019  13:28         Cash  618.38                 4.761905   

     gross income  Rating   total        low  
0         26.1415     9.1   81.69  10.670000  
1          3.8200     9.6   20.28   3.056000  
2         16.2155     7.4   53.33   6.618571  
3         23.2880     8.4   66.22   7.277500  
4         30.2085     5.3   93.31  12.330000  
..            ...     ...     ...        ...  
995        2.0175     6.2   41.35  40.350000  
996       48.6900     4.4  107.38   9.738000  
997        1.5920     7.7   32.84  31.840000  
998        3.2910     4.1   66.82  65.820000  
999       30.9190     6.6   95.34  12.620000  

[1000 rows x 19 columns]

#### Renamed a 'Unit Price' variable as 'Quantity '

In [25]:
market.rename(columns={'Payment':'Payment method','Total':'Final'})

Invoice ID Branch       City Customer type  Gender  \
0    750-67-8428      A     Yangon        Member  Female   
1    226-31-3081      C  Naypyitaw        Normal  Female   
2    631-41-3108      A     Yangon        Normal    Male   
3    123-19-1176      A     Yangon        Member    Male   
4    373-73-7910      A     Yangon        Normal    Male   
..           ...    ...        ...           ...     ...   
995  233-67-5758      C  Naypyitaw        Normal    Male   
996  303-96-2227      B   Mandalay        Normal  Female   
997  727-02-1313      A     Yangon        Member    Male   
998  347-56-2442      A     Yangon        Normal    Male   
999  849-09-3807      A     Yangon        Member  Female   

               Product line  Unit price  Quantity   Tax 5%      Final  \
0         Health and beauty       74.69         7  26.1415   548.9715   
1    Electronic accessories       15.28         5   3.8200    80.2200   
2        Home and lifestyle       46.33         7  16.2155   340.5255   
3         Health and beauty       58.22         8  23.2880   489.0480   
4         Sports and travel       86.31         7  30.2085   634.3785   
..                      ...         ...       ...      ...        ...   
995       Health and beauty       40.35         1   2.0175    42.3675   
996      Home and lifestyle       97.38        10  48.6900  1022.4900   
997      Food and beverages       31.84         1   1.5920    33.4320   
998      Home and lifestyle       65.82         1   3.2910    69.1110   
999     Fashion accessories       88.34         7  30.9190   649.2990   

          Date   Time Payment method    cogs  gross margin percentage  \
0     1/5/2019  13:08        Ewallet  522.83                 4.761905   
1     3/8/2019  10:29           Cash   76.40                 4.761905   
2     3/3/2019  13:23    Credit card  324.31                 4.761905   
3    1/27/2019  20:33        Ewallet  465.76                 4.761905   
4     2/8/2019  10:37        Ewallet  604.17                 4.761905   
..         ...    ...            ...     ...                      ...   
995  1/29/2019  13:46        Ewallet   40.35                 4.761905   
996   3/2/2019  17:16        Ewallet  973.80                 4.761905   
997   2/9/2019  13:22           Cash   31.84                 4.761905   
998  2/22/2019  15:33           Cash   65.82                 4.761905   
999  2/18/2019  13:28           Cash  618.38                 4.761905   

     gross income  Rating   total        low  
0         26.1415     9.1   81.69  10.670000  
1          3.8200     9.6   20.28   3.056000  
2         16.2155     7.4   53.33   6.618571  
3         23.2880     8.4   66.22   7.277500  
4         30.2085     5.3   93.31  12.330000  
..            ...     ...     ...        ...  
995        2.0175     6.2   41.35  40.350000  
996       48.6900     4.4  107.38   9.738000  
997        1.5920     7.7   32.84  31.840000  
998        3.2910     4.1   66.82  65.820000  
999       30.9190     6.6   95.34  12.620000  

[1000 rows x 19 columns]

#### Renamed Two variables 'Payment' as 'Payment method' and 'Total' as 'Final'

In [29]:
market.drop(['Tax 5%','Time'],axis=1)

Invoice ID Branch       City Customer type  Gender  \
0    750-67-8428      A     Yangon        Member  Female   
1    226-31-3081      C  Naypyitaw        Normal  Female   
2    631-41-3108      A     Yangon        Normal    Male   
3    123-19-1176      A     Yangon        Member    Male   
4    373-73-7910      A     Yangon        Normal    Male   
..           ...    ...        ...           ...     ...   
995  233-67-5758      C  Naypyitaw        Normal    Male   
996  303-96-2227      B   Mandalay        Normal  Female   
997  727-02-1313      A     Yangon        Member    Male   
998  347-56-2442      A     Yangon        Normal    Male   
999  849-09-3807      A     Yangon        Member  Female   

               Product line  Unit price  Quantity      Total       Date  \
0         Health and beauty       74.69         7   548.9715   1/5/2019   
1    Electronic accessories       15.28         5    80.2200   3/8/2019   
2        Home and lifestyle       46.33         7   340.5255   3/3/2019   
3         Health and beauty       58.22         8   489.0480  1/27/2019   
4         Sports and travel       86.31         7   634.3785   2/8/2019   
..                      ...         ...       ...        ...        ...   
995       Health and beauty       40.35         1    42.3675  1/29/2019   
996      Home and lifestyle       97.38        10  1022.4900   3/2/2019   
997      Food and beverages       31.84         1    33.4320   2/9/2019   
998      Home and lifestyle       65.82         1    69.1110  2/22/2019   
999     Fashion accessories       88.34         7   649.2990  2/18/2019   

         Payment    cogs  gross margin percentage  gross income  Rating  \
0        Ewallet  522.83                 4.761905       26.1415     9.1   
1           Cash   76.40                 4.761905        3.8200     9.6   
2    Credit card  324.31                 4.761905       16.2155     7.4   
3        Ewallet  465.76                 4.761905       23.2880     8.4   
4        Ewallet  604.17                 4.761905       30.2085     5.3   
..           ...     ...                      ...           ...     ...   
995      Ewallet   40.35                 4.761905        2.0175     6.2   
996      Ewallet  973.80                 4.761905       48.6900     4.4   
997         Cash   31.84                 4.761905        1.5920     7.7   
998         Cash   65.82                 4.761905        3.2910     4.1   
999         Cash  618.38                 4.761905       30.9190     6.6   

      total        low  
0     81.69  10.670000  
1     20.28   3.056000  
2     53.33   6.618571  
3     66.22   7.277500  
4     93.31  12.330000  
..      ...        ...  
995   41.35  40.350000  
996  107.38   9.738000  
997   32.84  31.840000  
998   66.82  65.820000  
999   95.34  12.620000  

[1000 rows x 17 columns]

#### We have dropped the variables 'Tax 5%' and 'Time' from the dataset. 

In [31]:
market.columns=[col.upper()for col in market.columns]

In [32]:
market

INVOICE ID BRANCH       CITY CUSTOMER TYPE  GENDER  \
0    750-67-8428      A     Yangon        Member  Female   
1    226-31-3081      C  Naypyitaw        Normal  Female   
2    631-41-3108      A     Yangon        Normal    Male   
3    123-19-1176      A     Yangon        Member    Male   
4    373-73-7910      A     Yangon        Normal    Male   
..           ...    ...        ...           ...     ...   
995  233-67-5758      C  Naypyitaw        Normal    Male   
996  303-96-2227      B   Mandalay        Normal  Female   
997  727-02-1313      A     Yangon        Member    Male   
998  347-56-2442      A     Yangon        Normal    Male   
999  849-09-3807      A     Yangon        Member  Female   

               PRODUCT LINE  UNIT PRICE  QUANTITY   TAX 5%      TOTAL  \
0         Health and beauty       74.69         7  26.1415   548.9715   
1    Electronic accessories       15.28         5   3.8200    80.2200   
2        Home and lifestyle       46.33         7  16.2155   340.5255   
3         Health and beauty       58.22         8  23.2880   489.0480   
4         Sports and travel       86.31         7  30.2085   634.3785   
..                      ...         ...       ...      ...        ...   
995       Health and beauty       40.35         1   2.0175    42.3675   
996      Home and lifestyle       97.38        10  48.6900  1022.4900   
997      Food and beverages       31.84         1   1.5920    33.4320   
998      Home and lifestyle       65.82         1   3.2910    69.1110   
999     Fashion accessories       88.34         7  30.9190   649.2990   

          DATE   TIME      PAYMENT    COGS  GROSS MARGIN PERCENTAGE  \
0     1/5/2019  13:08      Ewallet  522.83                 4.761905   
1     3/8/2019  10:29         Cash   76.40                 4.761905   
2     3/3/2019  13:23  Credit card  324.31                 4.761905   
3    1/27/2019  20:33      Ewallet  465.76                 4.761905   
4     2/8/2019  10:37      Ewallet  604.17                 4.761905   
..         ...    ...          ...     ...                      ...   
995  1/29/2019  13:46      Ewallet   40.35                 4.761905   
996   3/2/2019  17:16      Ewallet  973.80                 4.761905   
997   2/9/2019  13:22         Cash   31.84                 4.761905   
998  2/22/2019  15:33         Cash   65.82                 4.761905   
999  2/18/2019  13:28         Cash  618.38                 4.761905   

     GROSS INCOME  RATING   TOTAL        LOW  
0         26.1415     9.1   81.69  10.670000  
1          3.8200     9.6   20.28   3.056000  
2         16.2155     7.4   53.33   6.618571  
3         23.2880     8.4   66.22   7.277500  
4         30.2085     5.3   93.31  12.330000  
..            ...     ...     ...        ...  
995        2.0175     6.2   41.35  40.350000  
996       48.6900     4.4  107.38   9.738000  
997        1.5920     7.7   32.84  31.840000  
998        3.2910     4.1   66.82  65.820000  
999       30.9190     6.6   95.34  12.620000  

[1000 rows x 19 columns]

#### Displayed the Variables name of the dataset in the Uppercase,

In [33]:
market.columns=[col.lower()for col in market.columns]

In [34]:
market

invoice id branch       city customer type  gender  \
0    750-67-8428      A     Yangon        Member  Female   
1    226-31-3081      C  Naypyitaw        Normal  Female   
2    631-41-3108      A     Yangon        Normal    Male   
3    123-19-1176      A     Yangon        Member    Male   
4    373-73-7910      A     Yangon        Normal    Male   
..           ...    ...        ...           ...     ...   
995  233-67-5758      C  Naypyitaw        Normal    Male   
996  303-96-2227      B   Mandalay        Normal  Female   
997  727-02-1313      A     Yangon        Member    Male   
998  347-56-2442      A     Yangon        Normal    Male   
999  849-09-3807      A     Yangon        Member  Female   

               product line  unit price  quantity   tax 5%      total  \
0         Health and beauty       74.69         7  26.1415   548.9715   
1    Electronic accessories       15.28         5   3.8200    80.2200   
2        Home and lifestyle       46.33         7  16.2155   340.5255   
3         Health and beauty       58.22         8  23.2880   489.0480   
4         Sports and travel       86.31         7  30.2085   634.3785   
..                      ...         ...       ...      ...        ...   
995       Health and beauty       40.35         1   2.0175    42.3675   
996      Home and lifestyle       97.38        10  48.6900  1022.4900   
997      Food and beverages       31.84         1   1.5920    33.4320   
998      Home and lifestyle       65.82         1   3.2910    69.1110   
999     Fashion accessories       88.34         7  30.9190   649.2990   

          date   time      payment    cogs  gross margin percentage  \
0     1/5/2019  13:08      Ewallet  522.83                 4.761905   
1     3/8/2019  10:29         Cash   76.40                 4.761905   
2     3/3/2019  13:23  Credit card  324.31                 4.761905   
3    1/27/2019  20:33      Ewallet  465.76                 4.761905   
4     2/8/2019  10:37      Ewallet  604.17                 4.761905   
..         ...    ...          ...     ...                      ...   
995  1/29/2019  13:46      Ewallet   40.35                 4.761905   
996   3/2/2019  17:16      Ewallet  973.80                 4.761905   
997   2/9/2019  13:22         Cash   31.84                 4.761905   
998  2/22/2019  15:33         Cash   65.82                 4.761905   
999  2/18/2019  13:28         Cash  618.38                 4.761905   

     gross income  rating   total        low  
0         26.1415     9.1   81.69  10.670000  
1          3.8200     9.6   20.28   3.056000  
2         16.2155     7.4   53.33   6.618571  
3         23.2880     8.4   66.22   7.277500  
4         30.2085     5.3   93.31  12.330000  
..            ...     ...     ...        ...  
995        2.0175     6.2   41.35  40.350000  
996       48.6900     4.4  107.38   9.738000  
997        1.5920     7.7   32.84  31.840000  
998        3.2910     4.1   66.82  65.820000  
999       30.9190     6.6   95.34  12.620000  

[1000 rows x 19 columns]

#### Displayed the Variables name of the dataset in the lowercase.

In [37]:
market.sort_values(by=['unit price','quantity'],ascending=[True,False])

invoice id branch       city customer type  gender         product line  \
944  333-23-2632      A     Yangon        Member    Male    Health and beauty   
572  239-48-4278      A     Yangon        Member    Male   Food and beverages   
784  516-77-6464      C  Naypyitaw        Member  Female    Health and beauty   
822  784-21-9238      C  Naypyitaw        Member    Male    Sports and travel   
881  115-38-7388      C  Naypyitaw        Member  Female  Fashion accessories   
..           ...    ...        ...           ...     ...                  ...   
283  667-92-0055      A     Yangon        Member    Male    Health and beauty   
494  437-53-3084      B   Mandalay        Normal    Male  Fashion accessories   
930  641-62-7288      B   Mandalay        Normal    Male   Home and lifestyle   
122  219-22-9386      B   Mandalay        Member    Male    Sports and travel   
983  148-41-7930      C  Naypyitaw        Normal    Male    Health and beauty   

     unit price  quantity   tax 5%     total       date   time      payment  \
944       10.08         7   3.5280   74.0880  3/28/2019  20:14         Cash   
572       10.13         7   3.5455   74.4555  3/10/2019  19:35      Ewallet   
784       10.16         5   2.5400   53.3400  2/24/2019  13:08      Ewallet   
822       10.17         1   0.5085   10.6785   2/7/2019  14:15         Cash   
881       10.18         8   4.0720   85.5120  3/30/2019  12:51  Credit card   
..          ...       ...      ...       ...        ...    ...          ...   
283       99.83         6  29.9490  628.9290   3/4/2019  15:02      Ewallet   
494       99.89         2   9.9890  209.7690  2/26/2019  11:48      Ewallet   
930       99.92         6  29.9760  629.4960  3/24/2019  13:33      Ewallet   
122       99.96         9  44.9820  944.6220   3/9/2019  17:26  Credit card   
983       99.96         7  34.9860  734.7060  1/23/2019  10:33         Cash   

       cogs  gross margin percentage  gross income  rating   total        low  
944   70.56                 4.761905        3.5280     4.2   17.08   1.440000  
572   70.91                 4.761905        3.5455     8.3   17.13   1.447143  
784   50.80                 4.761905        2.5400     4.1   15.16   2.032000  
822   10.17                 4.761905        0.5085     5.9   11.17  10.170000  
881   81.44                 4.761905        4.0720     9.5   18.18   1.272500  
..      ...                      ...           ...     ...     ...        ...  
283  598.98                 4.761905       29.9490     8.5  105.83  16.638333  
494  199.78                 4.761905        9.9890     7.1  101.89  49.945000  
930  599.52                 4.761905       29.9760     7.1  105.92  16.653333  
122  899.64                 4.761905       44.9820     4.2  108.96  11.106667  
983  699.72                 4.761905       34.9860     6.1  106.96  14.280000  

[1000 rows x 19 columns]

#### In this we have sorted two variables 'unit price' and 'quantity' and then we have arranged the variable 'Unit Price' in the ascending order.

In [42]:
market.sort_values(by='quantity',ascending=True)

invoice id branch       city customer type  gender         product line  \
321  622-20-1945      C  Naypyitaw        Normal  Female    Health and beauty   
947  676-10-2200      B   Mandalay        Member    Male  Fashion accessories   
415  268-03-6164      B   Mandalay        Normal    Male    Health and beauty   
417  186-09-3669      C  Naypyitaw        Member  Female    Health and beauty   
88   504-35-8843      A     Yangon        Normal    Male    Sports and travel   
..           ...    ...        ...           ...     ...                  ...   
182  851-28-6367      A     Yangon        Member    Male    Sports and travel   
491  510-09-5628      A     Yangon        Member  Female  Fashion accessories   
493  148-82-2527      C  Naypyitaw        Member  Female   Home and lifestyle   
207  263-87-5680      C  Naypyitaw        Member  Female   Home and lifestyle   
882  311-13-6971      B   Mandalay        Member    Male    Sports and travel   

     unit price  quantity   tax 5%     total       date   time      payment  \
321       39.42         1   1.9710   41.3910  1/18/2019  15:08         Cash   
947       53.78         1   2.6890   56.4690   2/3/2019  20:13      Ewallet   
415       96.11         1   4.8055  100.9155  1/25/2019  16:28      Ewallet   
417       81.51         1   4.0755   85.5855  1/22/2019  10:57      Ewallet   
88        42.47         1   2.1235   44.5935   1/2/2019  16:57         Cash   
..          ...       ...      ...       ...        ...    ...          ...   
182       15.50        10   7.7500  162.7500  3/23/2019  10:55      Ewallet   
491       19.66        10   9.8300  206.4300  3/15/2019  18:20  Credit card   
493       12.12        10   6.0600  127.2600   3/5/2019  13:44  Credit card   
207       28.53        10  14.2650  299.5650  3/18/2019  17:38      Ewallet   
882       31.99        10  15.9950  335.8950  2/20/2019  15:18  Credit card   

       cogs  gross margin percentage  gross income  rating  total     low  
321   39.42                 4.761905        1.9710     8.4  40.42  39.420  
947   53.78                 4.761905        2.6890     4.7  54.78  53.780  
415   96.11                 4.761905        4.8055     7.8  97.11  96.110  
417   81.51                 4.761905        4.0755     9.2  82.51  81.510  
88    42.47                 4.761905        2.1235     5.7  43.47  42.470  
..      ...                      ...           ...     ...    ...     ...  
182  155.00                 4.761905        7.7500     8.0  25.50   1.550  
491  196.60                 4.761905        9.8300     7.2  29.66   1.966  
493  121.20                 4.761905        6.0600     8.4  22.12   1.212  
207  285.30                 4.761905       14.2650     7.8  38.53   2.853  
882  319.90                 4.761905       15.9950     9.9  41.99   3.199  

[1000 rows x 19 columns]

#### Sorted the variable named 'quantity' and arranged its values in ascending order.

In [45]:
market.sort_values(by='date')

invoice id branch       city customer type  gender  \
496  556-97-7101      C  Naypyitaw        Normal  Female   
484  493-65-6248      C  Naypyitaw        Member  Female   
567  651-88-7328      A     Yangon        Normal  Female   
523  133-14-7229      C  Naypyitaw        Normal    Male   
970  746-04-1077      B   Mandalay        Member  Female   
..           ...    ...        ...           ...     ...   
393  308-47-4913      A     Yangon        Member  Female   
73   841-35-6630      C  Naypyitaw        Normal  Female   
433  616-87-0016      B   Mandalay        Normal    Male   
585  127-47-6963      A     Yangon        Normal    Male   
84   360-39-5055      C  Naypyitaw        Member    Male   

               product line  unit price  quantity   tax 5%     total  \
496  Electronic accessories       63.22         2   6.3220  132.7620   
484       Sports and travel       36.98        10  18.4900  388.2900   
567     Fashion accessories       65.74         9  29.5830  621.2430   
523       Health and beauty       62.87         2   6.2870  132.0270   
970      Food and beverages       84.63        10  42.3150  888.6150   
..                      ...         ...       ...      ...       ...   
393       Sports and travel       52.26        10  26.1300  548.7300   
73   Electronic accessories       75.91         6  22.7730  478.2330   
433     Fashion accessories       95.54         7  33.4390  702.2190   
585       Health and beauty       51.71         4  10.3420  217.1820   
84        Sports and travel       48.91         5  12.2275  256.7775   

         date   time      payment    cogs  gross margin percentage  \
496  1/1/2019  15:51         Cash  126.44                 4.761905   
484  1/1/2019  19:48  Credit card  369.80                 4.761905   
567  1/1/2019  13:55         Cash  591.66                 4.761905   
523  1/1/2019  11:43         Cash  125.74                 4.761905   
970  1/1/2019  11:36  Credit card  846.30                 4.761905   
..        ...    ...          ...     ...                      ...   
393  3/9/2019  12:45  Credit card  522.60                 4.761905   
73   3/9/2019  18:21         Cash  455.46                 4.761905   
433  3/9/2019  14:36  Credit card  668.78                 4.761905   
585  3/9/2019  13:53  Credit card  206.84                 4.761905   
84   3/9/2019  10:17         Cash  244.55                 4.761905   

     gross income  rating   total        low  
496        6.3220     8.5   65.22  31.610000  
484       18.4900     7.0   46.98   3.698000  
567       29.5830     7.7   74.74   7.304444  
523        6.2870     5.0   64.87  31.435000  
970       42.3150     9.0   94.63   8.463000  
..            ...     ...     ...        ...  
393       26.1300     6.2   62.26   5.226000  
73        22.7730     8.7   81.91  12.651667  
433       33.4390     9.6  102.54  13.648571  
585       10.3420     9.8   55.71  12.927500  
84        12.2275     6.6   53.91   9.782000  

[1000 rows x 19 columns]

#### We have sorted the variable named 'date'.

In [46]:
market.drop(['low'],axis=1,inplace=True)

In [47]:
market

invoice id branch       city customer type  gender  \
0    750-67-8428      A     Yangon        Member  Female   
1    226-31-3081      C  Naypyitaw        Normal  Female   
2    631-41-3108      A     Yangon        Normal    Male   
3    123-19-1176      A     Yangon        Member    Male   
4    373-73-7910      A     Yangon        Normal    Male   
..           ...    ...        ...           ...     ...   
995  233-67-5758      C  Naypyitaw        Normal    Male   
996  303-96-2227      B   Mandalay        Normal  Female   
997  727-02-1313      A     Yangon        Member    Male   
998  347-56-2442      A     Yangon        Normal    Male   
999  849-09-3807      A     Yangon        Member  Female   

               product line  unit price  quantity   tax 5%      total  \
0         Health and beauty       74.69         7  26.1415   548.9715   
1    Electronic accessories       15.28         5   3.8200    80.2200   
2        Home and lifestyle       46.33         7  16.2155   340.5255   
3         Health and beauty       58.22         8  23.2880   489.0480   
4         Sports and travel       86.31         7  30.2085   634.3785   
..                      ...         ...       ...      ...        ...   
995       Health and beauty       40.35         1   2.0175    42.3675   
996      Home and lifestyle       97.38        10  48.6900  1022.4900   
997      Food and beverages       31.84         1   1.5920    33.4320   
998      Home and lifestyle       65.82         1   3.2910    69.1110   
999     Fashion accessories       88.34         7  30.9190   649.2990   

          date   time      payment    cogs  gross margin percentage  \
0     1/5/2019  13:08      Ewallet  522.83                 4.761905   
1     3/8/2019  10:29         Cash   76.40                 4.761905   
2     3/3/2019  13:23  Credit card  324.31                 4.761905   
3    1/27/2019  20:33      Ewallet  465.76                 4.761905   
4     2/8/2019  10:37      Ewallet  604.17                 4.761905   
..         ...    ...          ...     ...                      ...   
995  1/29/2019  13:46      Ewallet   40.35                 4.761905   
996   3/2/2019  17:16      Ewallet  973.80                 4.761905   
997   2/9/2019  13:22         Cash   31.84                 4.761905   
998  2/22/2019  15:33         Cash   65.82                 4.761905   
999  2/18/2019  13:28         Cash  618.38                 4.761905   

     gross income  rating   total  
0         26.1415     9.1   81.69  
1          3.8200     9.6   20.28  
2         16.2155     7.4   53.33  
3         23.2880     8.4   66.22  
4         30.2085     5.3   93.31  
..            ...     ...     ...  
995        2.0175     6.2   41.35  
996       48.6900     4.4  107.38  
997        1.5920     7.7   32.84  
998        3.2910     4.1   66.82  
999       30.9190     6.6   95.34  

[1000 rows x 18 columns]

#### We have dropped the variable named 'low' and used inplace function

In [48]:
market.groupby('unit price').count()

invoice id  branch  city  customer type  gender  product line  \
unit price                                                                  
10.08                1       1     1              1       1             1   
10.13                1       1     1              1       1             1   
10.16                1       1     1              1       1             1   
10.17                1       1     1              1       1             1   
10.18                1       1     1              1       1             1   
...                ...     ...   ...            ...     ...           ...   
99.82                2       2     2              2       2             2   
99.83                1       1     1              1       1             1   
99.89                1       1     1              1       1             1   
99.92                1       1     1              1       1             1   
99.96                2       2     2              2       2             2   

            quantity  tax 5%  total  date  time  payment  cogs  \
unit price                                                       
10.08              1       1      1     1     1        1     1   
10.13              1       1      1     1     1        1     1   
10.16              1       1      1     1     1        1     1   
10.17              1       1      1     1     1        1     1   
10.18              1       1      1     1     1        1     1   
...              ...     ...    ...   ...   ...      ...   ...   
99.82              2       2      2     2     2        2     2   
99.83              1       1      1     1     1        1     1   
99.89              1       1      1     1     1        1     1   
99.92              1       1      1     1     1        1     1   
99.96              2       2      2     2     2        2     2   

            gross margin percentage  gross income  rating  total  
unit price                                                        
10.08                             1             1       1      1  
10.13                             1             1       1      1  
10.16                             1             1       1      1  
10.17                             1             1       1      1  
10.18                             1             1       1      1  
...                             ...           ...     ...    ...  
99.82                             2             2       2      2  
99.83                             1             1       1      1  
99.89                             1             1       1      1  
99.92                             1             1       1      1  
99.96                             2             2       2      2  

[943 rows x 17 columns]

#### By Performing groupby function on Unit Price variable and used count function it is displays us this variable takes place in 943 observations.

In [49]:
market=pd.read_csv("C:/Users/anand/OneDrive/Desktop/super_market.csv")

In [50]:
market

Invoice ID Branch       City Customer type  Gender  \
0    750-67-8428      A     Yangon        Member  Female   
1    226-31-3081      C  Naypyitaw        Normal  Female   
2    631-41-3108      A     Yangon        Normal    Male   
3    123-19-1176      A     Yangon        Member    Male   
4    373-73-7910      A     Yangon        Normal    Male   
..           ...    ...        ...           ...     ...   
995  233-67-5758      C  Naypyitaw        Normal    Male   
996  303-96-2227      B   Mandalay        Normal  Female   
997  727-02-1313      A     Yangon        Member    Male   
998  347-56-2442      A     Yangon        Normal    Male   
999  849-09-3807      A     Yangon        Member  Female   

               Product line  Unit price  Quantity   Tax 5%      Total  \
0         Health and beauty       74.69         7  26.1415   548.9715   
1    Electronic accessories       15.28         5   3.8200    80.2200   
2        Home and lifestyle       46.33         7  16.2155   340.5255   
3         Health and beauty       58.22         8  23.2880   489.0480   
4         Sports and travel       86.31         7  30.2085   634.3785   
..                      ...         ...       ...      ...        ...   
995       Health and beauty       40.35         1   2.0175    42.3675   
996      Home and lifestyle       97.38        10  48.6900  1022.4900   
997      Food and beverages       31.84         1   1.5920    33.4320   
998      Home and lifestyle       65.82         1   3.2910    69.1110   
999     Fashion accessories       88.34         7  30.9190   649.2990   

          Date   Time      Payment    cogs  gross margin percentage  \
0     1/5/2019  13:08      Ewallet  522.83                 4.761905   
1     3/8/2019  10:29         Cash   76.40                 4.761905   
2     3/3/2019  13:23  Credit card  324.31                 4.761905   
3    1/27/2019  20:33      Ewallet  465.76                 4.761905   
4     2/8/2019  10:37      Ewallet  604.17                 4.761905   
..         ...    ...          ...     ...                      ...   
995  1/29/2019  13:46      Ewallet   40.35                 4.761905   
996   3/2/2019  17:16      Ewallet  973.80                 4.761905   
997   2/9/2019  13:22         Cash   31.84                 4.761905   
998  2/22/2019  15:33         Cash   65.82                 4.761905   
999  2/18/2019  13:28         Cash  618.38                 4.761905   

     gross income  Rating  
0         26.1415     9.1  
1          3.8200     9.6  
2         16.2155     7.4  
3         23.2880     8.4  
4         30.2085     5.3  
..            ...     ...  
995        2.0175     6.2  
996       48.6900     4.4  
997        1.5920     7.7  
998        3.2910     4.1  
999       30.9190     6.6  

[1000 rows x 17 columns]

In [54]:
print(market['Unit price'].isnull().sum())

0


#### We  found out that there is no null values present in the 'Unit Price'' variable.¶

In [55]:
print(market.columns.values)

['Invoice ID' 'Branch' 'City' 'Customer type' 'Gender' 'Product line'
 'Unit price' 'Quantity' 'Tax 5%' 'Total' 'Date' 'Time' 'Payment' 'cogs'
 'gross margin percentage' 'gross income' 'Rating']


#### Displayed the variables but by using print function.